src/main/scala/org/broadinstitute/dsde/firecloud/utils/PermissionsSupport.scala

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.dsde.firecloud.utils
+
+import org.broadinstitute.dsde.firecloud.dataaccess.RawlsDAO
+import org.broadinstitute.dsde.firecloud.{FireCloudException, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.firecloud.model.{UserInfo, WithAccessToken, errorReportSource}
+import org.broadinstitute.dsde.rawls.model.{ErrorReport, WorkspaceAccessLevels, WorkspaceResponse}
+import org.broadinstitute.dsde.firecloud.service.PerRequest.PerRequestMessage
+import spray.http.{HttpRequest, StatusCodes}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Created by dvoet on 11/5/15.
+ */
+trait PermissionsSupport {
+  protected val rawlsDAO: RawlsDAO
+  implicit protected val executionContext: ExecutionContext
+
+  def tryIsAdmin(userInfo: UserInfo): Future[Boolean] = {
+    rawlsDAO.isAdmin(userInfo) recoverWith { case t => throw new FireCloudException("Unable to query for admin status.", t) }
+  }
+
+  def asAdmin(op: => Future[PerRequestMessage])(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
+    tryIsAdmin(userInfo) flatMap { isAdmin =>
+      if (isAdmin) op else Future.failed(new FireCloudExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, "You must be an admin.")))
+    }
+  }
+
+  def tryIsCurator(userInfo: UserInfo): Future[Boolean] = {
+    rawlsDAO.isLibraryCurator(userInfo) recoverWith { case t => throw new FireCloudException("Unable to query for library curator status.", t) }
+  }
+
+  def asCurator(op: => Future[PerRequestMessage])(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
+    tryIsCurator(userInfo) flatMap { isCurator =>
+      if (isCurator) op else Future.failed(new FireCloudExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, "You must be a library curator.")))
+    }
+  }
+
+  def hasAccessOrAdmin(workspaceNamespace: String, workspaceName: String, neededLevel: WorkspaceAccessLevels.WorkspaceAccessLevel, userInfo: UserInfo): Future[Boolean] = {
+    tryIsAdmin(userInfo) flatMap { isadmin =>
+      if (!isadmin) {
+        rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)(userInfo.asInstanceOf[WithAccessToken]) map { ws =>
+          ws.accessLevel >= neededLevel
+        }
+      } else {
+        Future.successful(true)
+      }
+    }
+  }
+
```

Comment:
the three methods below this are not used anywhere in the code, according to IntelliJ. Do we still need them?